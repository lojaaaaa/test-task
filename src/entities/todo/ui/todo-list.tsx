
import { ITodo } from 'src/app/store';
import { TodoItem } from 'src/entities/todo';

interface TodoListProps {
  items: ITodo[];
}

export const TodoList = ({ items }: TodoListProps) => (
  <ul className="flex flex-col">
    {
      items.length
        ? items.map(({ id, value, isCompleted }) => (
          <TodoItem key={id} id={id} value={value} isCompleted={isCompleted} />
        ))
        : <span className="text-gray-500 text-center px-2 py-4">None</span>
    }
  </ul>
);
