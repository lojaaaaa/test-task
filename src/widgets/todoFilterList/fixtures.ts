export const filterOptions = [
  { key: '1', label: 'All' },
  { key: '2', label: 'Active' },
  { key: '3', label: 'Completed' }
];