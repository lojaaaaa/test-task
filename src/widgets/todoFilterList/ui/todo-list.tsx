import { filterOptions } from "../fixtures";

interface TodoFilterListProps {
  activeKeys: string[];
  toggleActiveKey: (key: string) => void;
};

export const TodoFilterList = ({ activeKeys, toggleActiveKey }: TodoFilterListProps) => (
  <ul className="flex gap-4">
    {filterOptions.map(({ key, label }) => (
      <li
        key={key}
        onClick={() => toggleActiveKey(key)}
        className={activeKeys.includes(key) ? 'cursor-pointer rounded-lg p-2 border border-red-200' : 'cursor-pointer p-2'}
      >
        {label}
      </li>
    ))}
  </ul>
);
