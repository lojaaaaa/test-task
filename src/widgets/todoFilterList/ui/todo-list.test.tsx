import { render, screen, fireEvent } from '@testing-library/react';
import { TodoFilterList } from './todo-list';
import { describe, it, expect, vi } from 'vitest';
import { filterOptions } from '../fixtures';

describe('TodoFilterList Component', () => {
  it('renders filter options correctly', () => {
    render(
      <TodoFilterList activeKey="1" setActiveKey={() => {}} />
    );

    filterOptions.forEach(({ label }) => {
      expect(screen.getByText(label)).toBeInTheDocument();
    });
  });

  it('highlights the active filter', () => {
    render(
      <TodoFilterList activeKey="2" setActiveKey={() => {}} />
    );

    const activeFilter = screen.getByText('Active');
    expect(activeFilter).toHaveClass('cursor-pointer rounded-lg p-2 border border-red-200');
  });

  it('calls setActiveKey when a filter is clicked', () => {
    const setActiveKeyMock = vi.fn();

    render(
      <TodoFilterList activeKey="1" setActiveKey={setActiveKeyMock} />
    );

    const filterToClick = screen.getByText('Completed');
    fireEvent.click(filterToClick);

    expect(setActiveKeyMock).toHaveBeenCalledWith('3');
  });

  
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <TodoFilterList activeKey="1" setActiveKey={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
