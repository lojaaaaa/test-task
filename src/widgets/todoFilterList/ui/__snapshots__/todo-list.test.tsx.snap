// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoFilterList Component > matches the snapshot 1`] = `
<DocumentFragment>
  <ul
    class="flex gap-4"
  >
    <li
      class="cursor-pointer rounded-lg p-2 border border-red-200"
    >
      All
    </li>
    <li
      class="cursor-pointer p-2"
    >
      Active
    </li>
    <li
      class="cursor-pointer p-2"
    >
      Completed
    </li>
  </ul>
</DocumentFragment>
`;
